src/pages/Dashboard/Dashboard.test.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react'
+import { fireEvent } from '@testing-library/react'
+import { mockNavigateTo } from 'react-router-dom'
+import { MockedProvider } from '@apollo/react-testing'
+import HolofuelUserQuery from 'graphql/HolofuelUserQuery.gql'
+import HappsQuery from 'graphql/HappsQuery.gql'
+import HolofuelLedgerQuery from 'graphql/HolofuelLedgerQuery.gql'
+import { presentHolofuelAmount } from 'utils'
+import { renderAndWait } from 'utils/test-utils'
+import { defaultHapp } from 'models/Happ'
+import Dashboard from './Dashboard'
+
+jest.mock('components/layout/PrimaryLayout')
+
+describe('Dashboard', () => {
+  it('renders empty states', async () => {
+    const mocks = [
+      {
+        request: {
+          query: HappsQuery
+        },
+        result: {
+          data: {
+            happs: []
+          }
+        }
+      },
+      {
+        request: {
+          query: HolofuelUserQuery
+        },
+        result: {
+          data: {
+            holofuelUser: {
+              id: '1',
+              nickname: ''
+            }
+          }
+        }
+      },
+      {
+        request: {
+          query: HolofuelLedgerQuery
+        },
+        result: {
+          data: {
+            holofuelLedger: {
+              balance: 0,
+              credit: 0,
+              payable: 0,
+              receivable: 0,
+              fees: 0
+            }
+          }
+        }
+      }
+    ]
+
+    const { getByText } = await renderAndWait(<MockedProvider mocks={mocks} addTypename={false}>
+      <Dashboard />
+    </MockedProvider>, 0)
+
+    expect(getByText('Hi!')).toBeInTheDocument()
+
+    const hosting = getByText('+ Host your first app')
+    expect(hosting).toBeInTheDocument()
+    fireEvent.click(hosting)
+    expect(mockNavigateTo).toHaveBeenCalledWith('/browse-happs')
```

Comment:
See my comments on `react-router-dom` mock at the top of the PR re this pattern. Here we can see the benefit we get of not having to actually render the router, and being able to easily test that the link does what we expect.